docs/grammar.txt

```diff
@@ -50,6 +50,7 @@ TypeName = ElementaryTypeName
          | Mapping
          | ArrayTypeName
          | FunctionTypeName
+         | ( 'address' 'payable' )
```

Comment:
Although this is part of ``ElementaryTypeName`` in the code, I think this is the easier and more accurate grammar change, since e.g. ``mapping(address payable => unit)`` and ``address payable(this)``, etc., will still not be valid.